src/Domain/Creators/TimeDependentOptions/BinaryCompactObject.cpp

```diff
@@ -329,102 +330,113 @@ void TimeDependentMapOptions<IsCylindrical>::build_maps(
-        // it's the same as the object center. If the transition ends at the
-        // sphere, then the center is the object center
-        const std::optional<std::array<double, 3>>& cube_center =
-            i == 0 ? cube_A_center : cube_B_center;
-        const std::array<double, 3>& inner_center = gsl::at(object_centers, i);
-        const std::array<double, 3>& outer_center =
-            transition_ends_at_cube
-                ? cube_center.value_or(gsl::at(object_centers, i))
-                : gsl::at(object_centers, i);
-
-        const double inner_sphericity = 1.0;
-        const double outer_sphericity = transition_ends_at_cube ? 0.0 : 1.0;
-
-        const double inner_radius = radii[0];
-        const double outer_radius =
-            transition_ends_at_cube ? radii[2] : radii[1];
-
-        using Wedge =
-            domain::CoordinateMaps::ShapeMapTransitionFunctions::Wedge;
-        for (size_t j = 0; j < axes.size(); j++) {
+      transition_func =
+          std::make_unique<domain::CoordinateMaps::ShapeMapTransitionFunctions::
+                               SphereTransition>(radii[0], radii[1]);
+
+      gsl::at(shape_maps_, i) = Shape{gsl::at(object_centers, i),
+                                      initial_l_max,
+                                      initial_l_max,
+                                      std::move(transition_func),
+                                      gsl::at(shape_names, i),
+                                      gsl::at(size_names, i)};
+    } else {
+      // These must match the order of orientations_for_sphere_wrappings() in
+      // DomainHelpers.hpp. The values must match that of Wedge::Axis
+      const std::array<int, 6> axes{3, -3, 2, -2, 1, -1};
+
+      const bool transition_ends_at_cube =
+          i == 0 ? shape_options_A_->transition_ends_at_cube
+                 : shape_options_B_->transition_ends_at_cube;
+
+      // These centers must take in to account if we have an offset of the
+      // center of the object and where the transition ends. The inner center
+      // is always the center of the object. The outer center depends on if we
+      // have an offset and where the transition ends. If the transition ends
+      // at the cube, then if we have an offset we use the cube center, if not
+      // it's the same as the object center. If the transition ends at the
+      // sphere, then the center is the object center
+      const std::optional<std::array<double, 3>>& cube_center =
+          i == 0 ? cube_A_center : cube_B_center;
+      const std::array<double, 3>& inner_center = gsl::at(object_centers, i);
+      const std::array<double, 3>& outer_center =
+          transition_ends_at_cube
+              ? cube_center.value_or(gsl::at(object_centers, i))
+              : gsl::at(object_centers, i);
+
+      const double inner_sphericity = 1.0;
+      const double outer_sphericity = transition_ends_at_cube ? 0.0 : 1.0;
```

Comment:
Can you add a comment here and/or below by the `inner_radius` about how if `filled` is True, then these values are only used for the outer shell of wedges and different values will be used for the inner wedges? I think this will help in the future when we (I) look at this again